authentication/MagentoAuthentication.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * This script is intended to handle Magento authentication via ZAP.
+ *
+ * Magento login form uses the anti-CSRF token called "form_key".
+ * The form key is generated once and used for the HTTP session.
+ * In order to log in we need to load the login page first to parse the form key
+ * and then send the form data along with the parsed form key.
+ *
+ * This script is based on the CasAuthentication script, so thank you guys for the example.
+ *
+ * @author Max Gopey <gopeyx@gmail.com>
+ */
+function getRequiredParamsNames(){
+    return ["loginUrl"];
+}
+
+function getOptionalParamsNames(){
+    return ["extraPostData"];
+}
+
+function getCredentialsParamsNames(){
+    return ["username", "password"];
+}
+
+function authenticate(helper, paramsValues, credentials) {
+    print("---- Magento authentication script has started ----");
```

Comment:
Yeah, that's fine. A local variable (ZAP does not provide anything to control that).